script.js

```diff
@@ -1,4 +1,227 @@
+var minCombined = ``; // to find min combined number among the dice rolls of a single player. Refreshes every player
+var arrayCombined = []; // to store all players combined number for that round. Refreshes every round
+
+var winningPlayer = 1; // player number. assign winning player each round
+var challenger = 2; // player number. assign challenger player each round (will +1)
+var arrayCombined = []; // to store both current winner [0] & challenger [1] combined number for that round. Refreshes every round
+
+var enterDiceAgainStatement = `Enter number of dice again!`;
+
+// function to roll X dices
+var generateXDiceRoll = function () {
+  var diceCounter = 0;
+  while (diceCounter < numberOfDice) {
+    arrayEachDice.push(generateRandomDiceRoll());
+    diceCounter += 1;
+  }
+};
+
+// function to generate BIGGEST combined number from dice rolls
+// 1. to loop through the arrayEachDice[] to find biggest number
+// 2. remove this biggest number from arrayEachDice[]
+// 3. find next biggest number from remaining elements in arrayEachDice[] and combine it the back of the previous biggest number -> maxCombined
+// 4. loop till arrayEachDice[] is empty
+// 5. push maxCombined into arrayCombined[]
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/max
+var generateBiggestCombined = function () {
+  var filterCounter = 0;
+  while (filterCounter < numberOfDice) {
+    maxNumber = arrayEachDice.reduce(function (a, b) {
+      return Math.max(a, b);
+    }, 0);
+    maxCombined = maxCombined + `` + maxNumber;
+    arrayEachDice.splice(arrayEachDice.indexOf(maxNumber), 1);
+    filterCounter += 1;
+  }
+  arrayCombined.push(maxCombined);
+};
+
+// function to generate SMALLEST combined number from dice rolls
+// works the same as generateBiggestCombined() but to find min combined
+// https://medium.com/@vladbezden/how-to-get-min-or-max-of-an-array-in-javascript-1c264ec6e1aa
+var generateSmallestCombined = function () {
+  var filterCounter = 0;
+  while (filterCounter < numberOfDice) {
+    minNumber = Math.min.apply(Math, arrayEachDice);
+    minCombined = minCombined + `` + minNumber;
+    arrayEachDice.splice(arrayEachDice.indexOf(minNumber), 1);
+    filterCounter += 1;
+  }
+  arrayCombined.push(minCombined);
+};
+
+// function to generate one game for X number of players
+var rollXDiceXPlayer = function () {
+  var playerCounter = 0;
+  var message = ``;
+
+  // every playerCounter is one player's round
+  // loop TWICE only since knock out matches are only between two players
+  while (playerCounter < 2) {
+    var xDiceRoll = generateXDiceRoll();
+    if (normalOrReverse == `normal`) {
+      var biggestCombined = generateBiggestCombined();
+    } else if (normalOrReverse == `reverse`) {
+      var smallestCombined = generateSmallestCombined();
+    }
+    maxCombined = ``;
+    minCombined = ``;
```

Comment:
are these variable declarations? if yes, we need to use the var keyword to declare a variable.  If you are just reassigning the variable to a new value then it's ok.